crates/config/src/toml.rs

```diff
@@ -777,6 +833,8 @@ mod tests {
 
     #[test]
     fn test_load_config_missing_include() -> Result<(), Box<dyn std::error::Error>> {
+        build_plugins()?;
```

Comment:
Generally in agreement. It's probably a little too quick-and-dirty as a solution.

Full context is that the config functions intentionally bail early if a plugin referenced by a config is missing. We don't want the program to crash from a missing plugin only after receiving an incoming request because then a blue/green style deployment might succeed when it should have failed, causing an outage that should have been trivial to prevent. But that now means our tests need the referenced plugins to be present and valid even for simple test cases we might otherwise want to fake.

Reply:
I took a look at the `test-context` crate as one potential option for solving this. It seems like the traits don't use a `Result` type, so I think I'd have to panic on a failed plugin build to get the behavior I'm looking for. Not sure the abstraction's worth having the extra dev dependency. There's also `test-harness` but that seems even more awkward.